Fix "delete" annotation button

gnodar01 opened the pull request:
- remove deleted annotation from active annotation ids so that the Transformer does not receive udnefined and throw an error
- resolves #473

gnodar01 committed:
[fix] Fix delete annotation

- remove deleted annotation from active annotation ids so that the
  Transformer does not receive udnefined and throw an error
- resolves #473

Andrea-Papaleo reviewed (commented):
I think a more general approach would be to add a listener akin to the one for `addAnnotation` in _dataListeners.ts_; it would listen for `deleteAnnotation` and dispatch `removeActiveAnnotationId`.

gnodar01 committed:
[mod, opt] Add delete annotation listener

removes the necessity to always dispatch removeActiveAnnotation[s]
before deleteAnnotation[s]
